tests/utils.py

```diff
@@ -549,3 +549,25 @@ def mocked_aws_s3_bucket_test_file(bucket_name, file_name, file_content="Test fi
         tmp_file.flush()
         s3.upload_file(Bucket=bucket_name, Filename=tmp_file.name, Key=file_name)
     return "s3://{}/{}".format(bucket_name, file_name)
+
+
+def mocked_http_test_file(dirname, file_url, file_content="This is a generated file for http test"):
+    # type: (str,str,str) -> str
+    """
+    Generates a test file reference from dummy data
+    """
+    path = dirname+file_url.split("localhost")[-1]
+    with open(path, "w") as tmp_file:
+        tmp_file.write(file_content)
+    return path
```

Comment:
This doesn't feel right. HTTP is not actually involved anywhere in this. 

The `http://` URL you provide as input in the test gets replaced right here, so the job execution that receives the returned path is a local file path, which is not different that the other case that gives directly `file://`.